pages/hydra/developer_resources/workflow_and_mediated_deposit/hyrax_1.0/workflow_and_mediated_deposit.md

```diff
@@ -107,14 +107,78 @@ w.setup
 ```
 Putting this code in `db/seeds.rb` means it will be called at the end of `bin/setup`, and you can call it anytime via `rake db:seed`
 
+## How do I test that all of this is working correctly?
+
+One approach is to make a capybara feature test for your approval workflow. You can log in
+as different users, and have those users perform workflow roles. Here is an example:
+```ruby
+# Check the ETD was assigned the right workflow
+etd = Etd.where(title: [title]).first
+expect(etd.active_workflow.name).to eq "emory_one_step_approval"
+expect(etd.to_sipity_entity.reload.workflow_state_name).to eq "pending_approval"
+
+# Check workflow permissions for depositing user
+available_workflow_actions = Hyrax::Workflow::PermissionQuery.scope_permitted_workflow_actions_available_for_current_state(user: user, entity: etd.to_sipity_entity).pluck(:name)
+expect(available_workflow_actions.include?("mark_as_reviewed")).to eq false
+expect(available_workflow_actions.include?("approve")).to eq false
+expect(available_workflow_actions.include?("request_changes")).to eq false
+expect(available_workflow_actions.include?("comment_only")).to eq false
+expect(available_workflow_actions.include?("hide")).to eq false
+expect(available_workflow_actions.include?("unhide")).to eq false
+
+# Check notifications for depositing user
+visit("/notifications?locale=en")
+expect(page).to have_content "#{title} (#{etd.id}) was deposited by #{user.email} and is awaiting approval."
+
+# Check notifications for approving user
+logout
+approving_user = User.where(email: "candleradmin@emory.edu").first
+login_as approving_user
+visit("/notifications?locale=en")
+expect(page).to have_content "#{title} (#{etd.id}) was deposited by #{user.email} and is awaiting approval."
+
+# Check workflow permissions for approving user
+available_workflow_actions = Hyrax::Workflow::PermissionQuery.scope_permitted_workflow_actions_available_for_current_state(user: approving_user, entity: etd.to_sipity_entity).pluck(:name)
+expect(available_workflow_actions.include?("mark_as_reviewed")).to eq false # this workflow step should only exist for Laney
```

Comment:
You may want to make these more concise by using `contain_exactly` https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/contain-exactly-matcher